src/main/drivers/barometer/barometer_bmp388.c

```diff
@@ -0,0 +1,320 @@
+    uint8_t intCtrlValue = (1 << BMP388_INT_DRDY_EN_BIT) | (0 << BMP388_INT_FFULL_EN_BIT) | (0 << BMP388_INT_FWTM_EN_BIT) | (0 << BMP388_INT_LATCH_BIT) | (1 << BMP388_INT_LEVEL_BIT) | (0 << BMP388_INT_OD_BIT);
+    busWriteRegister(busdev, BMP388_INT_CTRL_REG, intCtrlValue);
+#endif
+
+    busReadRegisterBuffer(busdev, BMP388_TRIMMING_NVM_PAR_T1_LSB_REG, (uint8_t *)&bmp388_cal, sizeof(bmp388_calib_param_t));
+
+
+    // set oversampling
+    busWriteRegister(busdev, BMP388_OSR_REG,
+        ((BMP388_PRESSURE_OSR << BMP388_OSR_P_BIT) & BMP388_OSR_P_MASK) |
+        ((BMP388_TEMPERATURE_OSR << BMP388_OSR4_T_BIT) & BMP388_OSR4_T_MASK)
+    );
+
+    bmp388BeginForcedMeasurement(busdev);
+
+    // these are dummy as temperature is measured as part of pressure
+    baro->ut_delay = 0;
+    baro->get_ut = bmp388_get_ut;
+    baro->start_ut = bmp388_start_ut;
+    // only _up part is executed, and gets both temperature and pressure
+    baro->start_up = bmp388_start_up;
+    baro->get_up = bmp388_get_up;
+
+    // See datasheet 3.9.2 "Measurement rate in forced mode and normal mode"
+    baro->up_delay = 234 +
+        (392 + (powerf(2, BMP388_PRESSURE_OSR + 1) * 2000)) +
+        (313 + (powerf(2, BMP388_TEMPERATURE_OSR + 1) * 2000));
+
+    baro->calculate = bmp388_calculate;
+
+    return true;
+}
+
+static void bmp388_start_ut(baroDev_t *baro)
+{
+    UNUSED(baro);
+    // dummy
+}
+
+static void bmp388_get_ut(baroDev_t *baro)
+{
+    UNUSED(baro);
+    // dummy
+}
+
+static void bmp388_start_up(baroDev_t *baro)
+{
+    // start measurement
+    bmp388BeginForcedMeasurement(&baro->busdev);
+}
+
+static void bmp388_get_up(baroDev_t *baro)
+{
+    uint8_t dataFrame[BMP388_DATA_FRAME_LENGTH];
+
+    // read data from sensor
+    busReadRegisterBuffer(&baro->busdev, BMP388_DATA_0_REG, dataFrame, BMP388_DATA_FRAME_LENGTH);
+
+    bmp388_up = ((uint32_t)(dataFrame[0]) << 0) | ((uint32_t)(dataFrame[1]) << 8) | ((uint32_t)(dataFrame[2]) << 16);
+    bmp388_ut = ((uint32_t)(dataFrame[3]) << 0) | ((uint32_t)(dataFrame[4]) << 8) | ((uint32_t)(dataFrame[5]) << 16);
+}
+
+// Returns temperature in DegC, resolution is 0.01 DegC. Output value of "5123" equals 51.23 DegC
+static int64_t bmp388_compensate_T(uint32_t uncomp_temperature)
+{
+    uint64_t partial_data1;
+    uint64_t partial_data2;
+    uint64_t partial_data3;
+    int64_t partial_data4;
+    int64_t partial_data5;
+    int64_t partial_data6;
+    int64_t comp_temp;
+
+    partial_data1 = uncomp_temperature - (256 * bmp388_cal.T1);
+    partial_data2 = bmp388_cal.T2 * partial_data1;
+    partial_data3 = partial_data1 * partial_data1;
+    partial_data4 = (int64_t)partial_data3 * bmp388_cal.T3;
+    partial_data5 = ((int64_t)(partial_data2 * 262144) + partial_data4);
+    partial_data6 = partial_data5 / 4294967296;
+    /* Update t_lin, needed for pressure calculation */
+    t_lin = partial_data6;
+    comp_temp = (int64_t)((partial_data6 * 25)  / 16384);
+
+#ifdef DEBUG_BARO_BMP388
```

Comment:
This should not be done. If you want to leave debug logging in, define a mode for it.

Reply:
the issue with defining a mode for it is that it adds to the flash size and adds a cli option.

Reply:
Ok, then the debug code should be removed.

Reply:
done